.github/workflows/site-cd-workflow.yml

```diff
@@ -0,0 +1,117 @@
+name: Site Deployment
+
+on:
+  workflow_call:
+    inputs:
+      working-directory:
+        required: true
+        type: string
+
+permissions:
+  id-token: write
+  contents: read
+
+env:
+  ARM_CLIENT_ID: ${{ secrets.AZURE_CLIENT_ID }}
+  ARM_SUBSCRIPTION_ID: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+  ARM_TENANT_ID: ${{ secrets.AZURE_TENANT_ID }}
+  ARM_USE_OIDC: true
+  TF_VAR_subscriptionId: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+  TF_VAR_localAdminUser: ${{ secrets.localAdminUser }}
+  TF_VAR_localAdminPassword: ${{ secrets.localAdminPassword }}
+  TF_VAR_domainAdminUser: ${{ secrets.domainAdminUser }}
+  TF_VAR_domainAdminPassword: ${{ secrets.domainAdminPassword }}
+  TF_VAR_deploymentUserPassword: ${{ secrets.deploymentUserPassword }}
+  TF_VAR_servicePrincipalId: ${{ secrets.servicePrincipalId }}
+  TF_VAR_servicePrincipalSecret: ${{ secrets.servicePrincipalSecret }}
+  TF_VAR_rpServicePrincipalObjectId: ${{ secrets.rpServicePrincipalObjectId }}
+  TF_VAR_vmAdminPassword: ${{ secrets.vmAdminPassword }}
+  TF_VAR_domainJoinPassword: ${{ secrets.domainJoinPassword }}
+
+jobs:
+  terraform:
+    name: "Terraform"
+    # runs-on: [windows-latest]
+    runs-on: [self-hosted]
```

Comment:
## Arc IaC Automation generated
Next steps: 
 - If you use GitHub runner, you need to delete this line(runs-on: [self-hosted]), uncomment the line(runs-on: [windows-latest]) and then resolve this comment
 - If you use self-hosted runner, just resolve this comment:)

If you have any problem, please file an [issue](https://github.com/Azure/Edge-infrastructure-quickstart-template/issues/new/choose) or contact arcIaCSupport@microsoft.com